google-beta/resource_pubsub_subscription.go

```diff
@@ -1,228 +1,369 @@
+// ----------------------------------------------------------------------------
+//
+//     ***     AUTO GENERATED CODE    ***    AUTO GENERATED CODE     ***
+//
+// ----------------------------------------------------------------------------
+//
+//     This file is automatically generated by Magic Modules and manual
+//     changes will be clobbered when the file is regenerated.
+//
+//     Please read more about how to change this file in
+//     .github/CONTRIBUTING.md.
+//
+// ----------------------------------------------------------------------------
+
 package google
 
 import (
 	"fmt"
+	"log"
+	"reflect"
+	"regexp"
+	"strconv"
+	"time"
 
 	"github.com/hashicorp/terraform/helper/schema"
-	"google.golang.org/api/pubsub/v1"
-	"regexp"
 )
 
+func getComputedTopicName(project, topic string) string {
+	match, _ := regexp.MatchString("projects\\/.*\\/topics\\/.*", topic)
+	if match {
+		return topic
+	}
+	return fmt.Sprintf("projects/%s/topics/%s", project, topic)
+}
+
+func getComputedSubscriptionName(project, subscription string) string {
+	match, _ := regexp.MatchString("projects\\/.*\\/subscriptions\\/.*", subscription)
+	if match {
+		return subscription
+	}
+	return fmt.Sprintf("projects/%s/subscriptions/%s", project, subscription)
+}
+
 func resourcePubsubSubscription() *schema.Resource {
 	return &schema.Resource{
 		Create: resourcePubsubSubscriptionCreate,
 		Read:   resourcePubsubSubscriptionRead,
-		Update: resourcePubsubSubscriptionUpdate,
 		Delete: resourcePubsubSubscriptionDelete,
 
 		Importer: &schema.ResourceImporter{
-			State: resourcePubsubSubscriptionStateImporter,
+			State: resourcePubsubSubscriptionImport,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(240 * time.Second),
+			Delete: schema.DefaultTimeout(240 * time.Second),
 		},
 
 		Schema: map[string]*schema.Schema{
 			"name": {
```

Comment:
My apologies in advance, new style `name` values are super fiddly when we allow short names to be used, and this resource started in a not-great place where you could use full or short names (and we need to preserve existing behaviours). I think? I've covered everything but there could very well be another pass just to fiddle with it.